components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminService.java

```diff
@@ -504,12 +525,31 @@ public OAuthRevocationResponseDTO revokeAuthzForAppsByResoureOwner(
             revokeRespDTO.setError(true);
             revokeRespDTO.setErrorCode(OAuth2ErrorCodes.INVALID_REQUEST);
             revokeRespDTO.setErrorMsg("Invalid revocation request");
+
+            //passing a single element array with null element to make sure listeners are triggered at least once
+            triggerPostRevokeListeners(oauthListeners, revokeRequestDTO, revokeRespDTO, new AccessTokenDO[]{null});
             return revokeRespDTO;
         }
         return new OAuthRevocationResponseDTO();
     }
 
+    private void triggerPostRevokeListeners(List<OauthEventListener> oauthListeners,
```

Comment:
We can go for a seperate method to trigger pre listeners too. Isn't it? That way code would be more consistent..